### DISCUSSION TITLE
开发 idea 插件需要学习哪些内容呢

### DISCUSSION THREAD

**Discussion by lin-mt** [YiiGuxing/TranslationPlugin#1658] [Q&A] [answered]
目前想学习开发 idea 插件，但是看了官网文档，感觉内容不少，但是找不到自己想要的内容，每次都是网上搜，这样感觉效率有点低，比如
<img width="694" alt="image" src="https://user-images.githubusercontent.com/58337359/175857227-1d49e096-1d1a-440c-a5a2-69a541a9cf4e.png">
这个扩展怎么加上去的
idea 的扩展点，你是怎么找到相关文档的
希望不吝赐教，感谢～
也希望给点学习 idea 插件开发相关的建议

**Comment by YiiGuxing** [comment, marked as answer]
开发 idea 插件主要是从官网文档和 idea 源代码中学习相关内容，对于我个人而言，源代码要占大部分，官网文档只是辅助。当需要开发某个功能，而切入的扩展点未知的情况下，我一般都是从源代码中找切入点的。比如你图中的扩展，在开发调时，我会从源代码上找到类似的IDE内置功能是怎么加去的，从而找到自己需要的那个切入点：
![img](https://user-images.githubusercontent.com/10737066/175871093-4a60b963-481a-4119-91be-66abb08e3550.png)
